### PR TITLE
[ut] add ut for cluster.go, clusterlease.go, membercluster_client.go

### PR DESCRIPTION
**Pull request by ikaven1024**
Signed-off-by: yingjinhui <yingjinhui@didiglobal.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
add ut for  pkg/util/cluster

**Which issue(s) this PR fixes**:
Part of #2579

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/util/cluster_test.go [hunk 1]
```diff
@@ -54,38 +59,84 @@ func TestCreateOrUpdateClusterObject(t *testing.T) {
 		{
 			name: "cluster exist, and update cluster",
 			args: args{
-				controlPlaneClient: fakeClient,
+				controlPlaneClient: karmadaclientsetfake.NewSimpleClientset(withAPIEndPoint(newCluster(ClusterMember1), "https://127.0.0.1:6443")),
 				clusterObj:         newCluster(ClusterMember1),
 				mutate: func(cluster *clusterv1alpha1.Cluster) {
 					cluster.Spec.SyncMode = clusterv1alpha1.Pull
 				},
-				aop: func() func() {
```
Comment (XiShanYongYe-Chang) [inline pkg/util/cluster_test.go:62]: The line 51 seems can be removed:
```
aop                func() func()
```

File: pkg/util/cluster_test.go [hunk 2]
```diff
@@ -153,3 +197,277 @@ func TestIsClusterIDUnique(t *testing.T) {
...
+			name: "cluster not found",
+			args: args{
+				hostClient:  fakeclient.NewClientBuilder().Build(),
+				clusterName: "test",
+			},
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			name: "cluster not found",
```
Comment (XiShanYongYe-Chang) [inline pkg/util/cluster_test.go:406]: This case name seems not correct.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks a lot!!

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**